### ISSUE TITLE
Counterfeiter generated fakes do not satisfy go lint.

### ISSUE THREAD

**Issue by robdimsdale** [maxbrunsfeld/counterfeiter#13]
Here's some sample output from a fake logger (let me know if you need more info but it should be clear enough to highlight the issue):

```
fakes/fake_logger.go:1:1: package comment should be of the form "Package fakes ..."
fakes/fake_logger.go:9:6: exported type FakeLogger should have comment or be unexported
fakes/fake_logger.go:17:1: exported method FakeLogger.LogLine should have comment or be unexported
fakes/fake_logger.go:28:1: exported method FakeLogger.LogLineCallCount should have comment or be unexported
fakes/fake_logger.go:34:1: exported method FakeLogger.LogLineArgsForCall should have comment or be unexported
```

**Comment by maxbrunsfeld** [comment]
Do you think it makes sense to add comments to the generated fakes just to silence this lint tool? It might make more sense to just not run `go lint` on your `fakes` packages. If you're interested in fixing this though, I'd merge your PR.

**Comment by robdimsdale** [comment]
Yeah currently I'm just not running go lint on the counterfeiter-generated fakes. I mostly agree that the comments aren't worth it - with the exception that they would appear in godoc and it might be useful for someone who's not familiar with the package to understand what it is. However I don't feel strongly about this.

Go lint does also throw at least one more interesting error:

```
if block ends with a return statement, so drop this else and outdent its block
```

It might be worth fixing this? Again I don't feel particularly strongly. We're only talking about pretty-printing, and the fakes package is only used (or at least should only be used) in testing.

I probably won't submit a PR so feel free to close this issue if you don't feel like this is worth doing.

**Comment by BooleanCat** [comment]
Your final point is solved by https://github.com/maxbrunsfeld/counterfeiter/pull/58

**Comment by tjarratt** [comment]
@robdimsdale @BooleanCat can we close this issue now that go has a standard for "generated code" that linters and other tools should ignore?

see: #66 for more details
